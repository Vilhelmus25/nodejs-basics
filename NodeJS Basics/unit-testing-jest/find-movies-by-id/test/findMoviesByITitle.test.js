/* eslint-disable func-call-spacing */
const findMoviesByTitle = require('../src/findMoviesByTitle')

test('findMoviesByTitle should be an array of movies objects', () => {
    // expect(findMoviesByTitle('F')).toEqual([
    //     { title: 'Flipper', emoji: 'dolphin' }      // ha van egy Flipper 2 a tömbben, akkor nem lesz jó, mert megadtam, hogy csak egyetlen ilyen objectet tartalmazzon a teszt szerint
    //     // ez azért nem jó, mert ha módosul a tömb, akkor a tesztet is módosítani kéne, ez rengeteg plusz munka lenne
    //     // erre jó a Snapshot teszt
    // ])

    expect.addSnapshotSerializer({
        test: ({ title, emoji }) => title && emoji,
        print: ({ title, emoji }) => `${emoji} ${title}`
    })
    expect(findMoviesByTitle('F')).toMatchSnapshot()        // így létrehoz egy __snapshot mappát
    // ha kitörlünk egy objektumot, akkor sír, mert nem egyezik a snapshot a valósággal
    // erre kell a jest --updateSnapshot parancs és frissíti a snapshotot és utána már jó lesz, egy pillanatkép = snapshot készül
})