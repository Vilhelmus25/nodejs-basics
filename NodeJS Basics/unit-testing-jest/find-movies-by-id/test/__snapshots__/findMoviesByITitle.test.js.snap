// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findMoviesByTitle should be an array of movies objects 1`] = `
Array [
  dolphin Flipper,
]
`;
